src/System.Private.CoreLib/src/Internal/Runtime/Augments/EnvironmentAugments.Unix.cs

```diff
@@ -30,7 +30,89 @@ private static void SetEnvironmentVariableCore(string variable, string value)
             if ("".Length != 0)
                 throw new NotImplementedException(); // Need to return something better than an empty environment block.
 
-            return Array.Empty<KeyValuePair<string,string>>();
+            unsafe
+            {
+                // Get byte** of environment variables from native interop
+                var unsafeBlock = (byte**)Interop.Sys.EnvironGetSystemEnvironment();
+                if (unsafeBlock == (byte**)0)
+                    throw new OutOfMemoryException();
+                
+                // Find total length of two-dimensional byte**
```

Comment:
It is not really necessary to copy over the whole thing into an array first.

Reply:
I went down this road first for obvious non-allocatey reasons, but learned something along the way: you can't `yield return` in an `unsafe` block. I guess that's why the Windows version does the same thing (copy out and then call separate function).

Reply:
If this works, it would be a key use of local functions. I'll give it a shot tomorrow.

Reply:
Done, and it works well.